pkg/scanners/clairify/clairify.go

```diff
@@ -411,12 +420,44 @@ func (c *clairify) GetVulnerabilities(image *storage.Image, components *clairGRP
 	return convertImageToImageScan(image.GetMetadata(), resp.GetImage()), nil
 }
 
+func retryOnGRPCErrors(ctx context.Context, name string, f func() error) error {
+	op := func() error {
+		err := f()
+		if err != nil {
+			e, _ := status.FromError(err)
+			switch e.Code() {
+			// Unavailable indicates the service is currently unavailable. The error code is
+			// set by the gRPC framework during failed connections, abrupt shutdown of a
+			// server process or network connection.
+			//
+			// Aborted is not be generated by the gRPC framework, but may be used by server
+			// to retry on certain conditions that are retriable, e.g concurrency issue,
+			// sequencer check failures, transaction aborts, etc.
+			case codes.Aborted, codes.Unavailable:
+				return err
+			default:
+				return backoff.Permanent(err)
+			}
+		}
+		return err
+	}
+	notify := func(err error, duration time.Duration) {
+		log.Warnf("calling %s() (retrying in %s): %v", name, duration, err)
+	}
+	eb := backoff.NewExponentialBackOff()
```

Comment:
Do you want to tweak defaults of this new exponential backoff? Defaults seem to allow 60 seconds of max delay and 15 minutes of max elapsed time. Neither looks too shiny in spite of the current 10 seconds limit.

Reply:
My intention was to use the defaults. In particular, the max elapsed time is irrelevant since we are using `*WithContext()`. Should we give up earlier than 10 seconds? That might be reasonable, specially if Scanner is out:  We would always try for 10s, where we could give up earlier. I can't see a problem with 10s at the moment. Clarify if you see any, please.

Reply:
I take back that giving up earlier might be possible. Typical calls to Scanner takes 4s according to experiments in an Openshift cluster, so shorter timeouts might increase the chances of us giving up calls that would succeed.